net/gatling/files/gatling.init

```diff
@@ -0,0 +1,114 @@
+#!/bin/sh /etc/rc.common
+
+# shellcheck shell=ash
+
+# Just looks for changes in the config-file and applies them with a
+# one-time-run.
+
+USE_PROCD=1
+# PROCD_DEBUG=1
+
+# busybox doesn't include the rev-command. Thus implementing it by ourself.
+rev() {
+    while read -r var; do
+        # function taken from https://stackoverflow.com/a/34668251
+        rev=""
+        i=1
+
+        while [ "$i" -le "${#var}" ]; do
+            rev="$(echo "$var" | awk -v i="$i" '{print(substr($0,i,1))}')$rev"
+            : $((i += 1))
+        done
+
+        echo "$rev"
+    done
+}
+
+# taken from /etc/init.d/uhttpd
+append_arg() {
+	local cfg="$1"
+	local var="$2"
+	local opt="$3"
+	local def="$4"
+	local val
+
+	config_get val "$cfg" "$var"
+	[ -n "$val" -o -n "$def" ] && procd_append_param command "$opt" "${val:-$def}"
+}
+
+service_triggers() {
+    procd_add_reload_trigger "gatling"
+}
+
+start_instance() {
+
+    local cfg="$1"
+	local listen http ftp_server
+	local enabled
+
+	config_get_bool enabled "$cfg" 'enabled' 1
+	[ $enabled -gt 0 ] || return
+
+    procd_open_instance
+
+    procd_set_param command /usr/bin/gatling
+    procd_set_param stdout 1
+    procd_set_param stderr 1
+    procd_set_param term_timeout 20
+
+    # get listen-address and slice it from back, to cut at port-delimiter
+    config_get http gatling listen_http
+    port=$(echo "$listen" | rev | cut -d':' -f 1 | rev )
```

Comment:
I do not see that the variable "listen" is set anywhere. It should be "http"?
Anyway, the extraction of a port here is rather fragile. How about using a separate host and port variable?

Reply:
```
procd_append_param command -i "[::]"
```
gives me the following error
```
Wed Apr 17 11:02:04 2024 daemon.err gatling[2551]: gatling: warning: could not parse IP address ::].
```
@BKPepe do you have an idea how to fix this?

Reply:
> I do not see that the variable "listen" is set anywhere. It should be "http"? Anyway, the extraction of a port here is rather fragile. How about using a separate host and port variable?

I chose this way, to stay somewhat compatible to the way, `uhttpd` does this (which I consider the reference webserver for OpenWrt).

I would be completely fine, to make separate options on this (one address, one port).

Reply:
We can keep it the way it is. But we can make it simplier with the lines in my comment.

Reply:
I added the lines you do suggest in your comments.

Reply:
After some tries I found out that gatling does not accept IPv6 addresses with "[]", you need to remove them:
````
ip="${listen_http%:*}"
ip="${ip:1:-1}"
````

Reply:
Thats a really cool thing. I added this in the recent version of the commit.